modules/custom/az_news/az_news_feeds/src/Form/AzNewsFeedsAdminForm.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace Drupal\az_news_feeds\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\migrate\MigrateExecutable;
+use Drupal\migrate\MigrateMessage;
+use Drupal\migrate\Plugin\MigrationPluginManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use GuzzleHttp\Client;
+
+/**
+ * Form for editing news importer source.
+ */
+class AzNewsFeedsAdminForm extends ConfigFormBase {
+
+  /**
+   * @var \Drupal\migrate\Plugin\MigrationPluginManager
+   */
+  protected $migrationPluginManager;
+
+  /**
+   * Constructs a AzNewsFeedsAdminForm object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\migrate\Plugin\MigrationPluginManager $migration_plugin_manager
+   *   Plugin manager for migration plugins.
+   */
+  public function __construct(
+    ConfigFactoryInterface $config_factory,
+    MigrationPluginManager $migration_plugin_manager
+    ) {
+    parent::__construct($config_factory);
+    $this->migrationPluginManager = $migration_plugin_manager;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('plugin.manager.migration')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'migrate_plus.migration_group.az_news_feeds',
+      'az_news_feeds.settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'az_news_feeds_admin';
+  }
+
+  /**
+   * Returns array of term options from UArizona News.
+   */
+  public function getRemoteTermOptions(): array {
+    $az_news_feeds_config = $this->config('az_news_feeds.settings');
+    $base_uri = $az_news_feeds_config->get('uarizona_news_category_base_uri');
+    $category_path = $az_news_feeds_config->get('uarizona_news_category_path');
+
+    $client = new Client([
```

Comment:
Nitpick:  would probably be preferable to inject the [core `http_client` service](https://api.drupal.org/api/drupal/core%21core.services.yml/service/http_client/9.2.x) to use here.